ibm/resource_ibm_app_config_environment.go

```diff
@@ -0,0 +1,258 @@
+/**
+ * (C) Copyright IBM Corp. 2021.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ibm
+
+import (
+	"fmt"
+
+	"github.com/IBM/appconfiguration-go-admin-sdk/appconfigurationv1"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceIbmAppConfigEnvironment() *schema.Resource {
+	return &schema.Resource{
+		Read:     resourceEnvironmentRead,
+		Create:   resourceEnvironmentCreate,
+		Update:   resourceEnvironmentUpdate,
+		Delete:   resourceEnvironmentDelete,
+		Importer: &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"guid": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "GUID of the App Configuration service. Get it from the service instance credentials section of the dashboard.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Environment name.",
+			},
+			"environment_id": {
```

Comment:
the resource is to create the environment how can environment_id a required fields here?

Reply:
this is correct. It's like when you fill the signup form with email and other details, later your email becomes your userId.